201-multi-vms-multi-nics/azuredeploy.json

```diff
@@ -0,0 +1,432 @@
+﻿{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "MultiNicStorageAccountName": {
+            "type": "string",
+            "metadata": {
+                "description": "Storageaccount name for the deployment - mainly used for VHD storage"
+            }
+        },
+        "AdminUserName": {
+            "type": "string",
+            "metadata": {
+                "description": "Username for the administrator login to the systems"
+            }
+        },
+        "AdminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Password for the administrator login to the systems"
+            }
+        },
+        "publicDnsName": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique public DNS prefix for the deployment. The fqdn will look something like '<dnsname>.westus.cloudapp.azure.com'. Up to 62 chars, digits or dashes, lowercase, should start with a letter: must conform to '^[a-z][a-z0-9-]{1,61}[a-z0-9]$'."
+            }
+        },
+        "rdpPort": {
+            "type": "int",
+            "defaultValue": 50001,
+            "metadata": {
+                "description": "Public port number for RDP"
+            }
+        }
+    },
+    "variables": {
+        "Virtual NetworkPrefix": "10.0.0.0/16",
+        "Virtual NetworkSubnetZoneAPrefix": "10.0.0.0/24",
+        "Virtual NetworkSubnetZoneBPrefix": "10.0.1.0/24",
+        "Virtual NetworkSubnetWebsPrefix": "10.0.2.0/24",
+        "Virtual NetworkSubnetOperationsPrefix": "10.0.3.0/24",
+        "FirewallsImagePublisher": "MicrosoftWindowsServer",
+        "FirewallsImageOffer": "WindowsServer",
+        "FirewallsOSDiskName": "FirewallsOSDisk",
+        "FirewallsVmSize": "Standard_D3",
+        "FirewallsWindowsOSVersion": "2012-R2-Datacenter",
+        "VnetID": "[resourceId('Microsoft.Network/virtualNetworks', 'MultiNICVNET')]",
+        "ZoneASubnetRef": "[concat(variables('VnetID'), '/subnets/', 'ZoneA')]",
+        "ZoneBSubnetRef": "[concat(variables('VnetID'), '/subnets/', 'ZoneB')]",
+        "FirewallsStorageAccountContainerName": "vhds",
+        "WebsImagePublisher": "MicrosoftWindowsServer",
+        "WebsImageOffer": "WindowsServer",
+        "WebsOSDiskName": "WebsOSDisk",
+        "WebsVmSize": "Standard_D2",
+        "WebsWindowsOSVersion": "2012-R2-Datacenter",
+        "WebsSubnetRef": "[concat(variables('VnetID'), '/subnets/', 'Webs')]",
+        "OperationsSubnetRef": "[concat(variables('VnetID'), '/subnets/', 'Operations')]",
+        "WebsStorageAccountContainerName": "vhds",
+        "OperationsVMSize": "Standard_D2",
+        "OperationsImagePublisher": "MicrosoftWindowsServer",
+        "OperationsImageOffer": "WindowsServer",
```

Comment:
Seems like there is an opportunity to reduce the number of variables since the Pub, Offer, SKU (called osversion here) is the same for firewalls, web and operations.